### PR TITLE
change common start signature

### PR DESCRIPTION
**Pull request by melekes**
Refs #45

### CODE REVIEW

**Comment by ebuchman** [comment]
I feel like Jae had a reason for wanting the bool here. We should check in with him.

**Comment by melekes** [comment]
> And should we add an error to OnStop?

I don't think we want to support that. like if there is an error in OnStop code, it should be ignored or dealt immediately? i.e. this should not bubble up

> Should we also do this for Reset ?

ye

File: common/service.go [hunk 1]
```diff
@@ -139,17 +146,17 @@ func (bs *BaseService) Stop() bool {
 func (bs *BaseService) OnStop() {}
 
 // Implements Service
-func (bs *BaseService) Reset() (bool, error) {
+func (bs *BaseService) Reset() error {
 	if !atomic.CompareAndSwapUint32(&bs.stopped, 1, 0) {
 		bs.Logger.Debug(Fmt("Can't reset %v. Not stopped", bs.name), "impl", bs.impl)
-		return false, nil
+		return fmt.Errorf("can't reset running %s", bs.name)
```
Comment (melekes) [inline common/service.go]: maybe we should `panic` here. like clearly if we are trying to reset running service, then something wrong with our code?

**Comment by ebuchman** [comment]
> I don't think we want to support that. like if there is an error in OnStop code, it should be ignored or dealt immediately? i.e. this should not bubble up

But we just changed `Stop` to return an `error`. Wouldn't it be misleading if it returned `nil` but some piece of the service didn't stop because `OnStop` had an error ? 

I agree it's not something that should happen, and OnStop should deal as best it can, but just trying to be consistent and unsurprising.

File: common/service.go (hunk 1 above)
Comment (ebuchman) [reply common/service.go]: We could, but it's probably best if we try to minimize panics in libraries ?

**Comment by adrianbrink** [comment]
I think it should bubble up from OnStop. A user should be able to catch an error and return it in his implementation of `OnStop`.

**Comment by melekes** [comment]
> but some piece of the service didn't stop because OnStop had an error ?

OnStop should never result in an error. One example could be Erlang gen_server http://erlang.org/doc/design_principles/gen_server_concepts.html#id64968 (notice that terminate always returns `ok`)

> but just trying to be consistent and unsurprising.

what if we add a comment about that?

**Comment by ebuchman** [comment]
> OnStop should never result in an error

What if it's supposed to do some cleanup and the cleanup fails ? I guess this means it's a completely contained responsibility. I suppose that's ok.

File: Makefile [hunk 2]
```diff
@@ -4,14 +4,18 @@ GOTOOLS = \
 	github.com/Masterminds/glide \
 	github.com/alecthomas/gometalinter
 
+PACKAGES=$(shell go list ./... | grep -v '/vendor/')
 REPO:=github.com/tendermint/tmlibs
 
 all: test
 
 NOVENDOR = go list github.com/tendermint/tmlibs/... | grep -v /vendor/
```
Comment (melekes) [inline Makefile]: this line can be removed now (it is identical to `PACKAGES`)

**Pull request merged**